modules/autocomplete-feed/main/index.coffee

```diff
@@ -0,0 +1,104 @@
+sortItems = (valueLookup) ->
+  valueLookup ?= hx.identity
+  (a, b) ->
+    if not a.disabled and b.disabled then -1
+    else if a.disabled and not b.disabled then 1
+    else hx.sort.compare(valueLookup(a), valueLookup(b))
+
+trimTrailingSpaces = (term) ->
+  newTerm = term
+  while newTerm.lastIndexOf(' ') is newTerm.length - 1
+    newTerm = newTerm.slice(0, newTerm.length - 1)
+  newTerm
+
+class AutocompleteFeed
+  constructor: (options = {}) ->
+    self = this
+
+    defaults =
+      filter: undefined
+      filterOptions: undefined
+      matchType: 'contains'
+      showOtherResults: false
+      trimTrailingSpaces: false
+      valueLookup: undefined
+
+    if options.valueLookup?
+      # default searchValue if valueLookup is defined
+      defaults.filterOptions =
+        searchValues: (datum) -> [self._.options.valueLookup(datum)]
+
+    resolvedOptions = hx.merge.defined defaults, options
+
+    # defined here so we can use the resolved options
+    resolvedOptions.filter ?= (items, term) ->
+      hx.filter[resolvedOptions.matchType](items, term, resolvedOptions.filterOptions)
+        .sort sortItems(resolvedOptions.valueLookup)
+
+    @_ =
+      options: resolvedOptions
+      resultsCache: new hx.Map
+
+
+  clearCache: ->
+    @_.resultsCache = new hx.Map
+    this
+
+  filter: (term = '', callback) ->
+    _ = @_
+    thisFilter = term + hx.randomId()
+    _.lastFilter = thisFilter
+
+    cacheitemsAndCallback = (results, otherResults = []) =>
+      if _.options.trimTrailingSpaces and results.length is 0 and term.lastIndexOf(' ') is term.length - 1
+        # The term has trailing spaces and no results were found
+        @filter(trimTrailingSpaces(term), callback)
+      else
+        # Cache the currently searched term items
+        _.resultsCache.set(term, {
+          results: results,
+          otherResults: otherResults
+        })
+        if thisFilter is _.lastFilter
+          callback(results, otherResults)
+
+    if _.options.matchType is 'external' and hx.isFunction(_.items)
+      # The matching is external so we don't filter here
+      _.items(term, cacheitemsAndCallback)
```

Comment:
Yes - it calls into `cacheItemsAndCallback` which is what AutoComplete did before (in a slightly more convoluted way)
The theory being if you're using external matching the search will return the same results for the same term. If that's not the case then calling `clearCache` in the external match function will provide the 'never cache' option. 
It might be worth adding an option called `noCache` or something along those lines?